### DISCUSSION TITLE
Bug: polling() not connecting to Telegram API (requests timeout)

### DISCUSSION THREAD

**Discussion by shorecodeorg** [eternnoir/pyTelegramBotAPI#2188] [General]
Please answer these questions before submitting your issue. Thanks!

1. What version of pyTelegramBotAPI are you using?

4.16.1

2. What OS are you using?

Linux Mint

3. What version of python are you using?

3.10.12

Issue: When I try to use polling() I get a timeout error from requests library.

Steps I took to isolate the issue: 

1.  Test outside Python
I tested the API endpoint using a curl tool and I was successful in getting responses from the getme and getupdates endpoint when using the same URL that pyTelegramBotAPI uses.

2.  Test requests
I tested the Python requests library with a different URL and received a quick 200 response.

3. Test requests with bot URL
I tested the Python requests library with the same URL that pyTelegramBotAPI uses and there was no response. Implemented a 10 second timeout throws the timeout error as expected.

4. Test requests with Telegram API
I tested the Python request library using the get method to connect to 'https://api.telegram.org/', again no response.

It would appear that the requests library cannot connect to the Telegram bot API.

When enabling debug I get the following:

2024-02-26 10:16:17,005 (apihelper.py:87 MainThread) DEBUG - TeleBot: "Request: method=get url=https://api.telegram.org/botXXXXXXX:{TOKEN}/getMe params=None files=None"

XXXXXXX being the first section before the colon of my api token, edited for privacy.

**Comment by denatuiumsaccharide** [comment]
@shorecodeorg 

The url is <ins>NOT like</ins>:
`https://api.telegram.org/botXXXXXXX:{TOKEN}/getMe` that you have specified above. In fact that XXXXXX is part of your token.

So its like: 
`https://api.telegram.org/bot{TOKEN}/getMe`

## With telebot:

import telebot
API_TOKEN = "your_API_here"
bot = telebot.TeleBot(API_TOKEN)
print(bot.get_updates()[0])
bot.infinity_polling()

## With requests:
import requests
url = "https://api.telegram.org/bot{TOKEN}"
bot = requests.get(url)
print(bot)
